src/kvstore/kvstore_dist_server.h

```diff
@@ -170,6 +187,33 @@ class KVStoreDistServer {
     app->Response(recved);
   }
 
+  void SetProfilerConfig(std::string params_str) {
+    std::vector<std::string> elems;
+    mxnet::kvstore::split(params_str, ',', std::back_inserter(elems));
+    std::vector<const char*> ckeys;
+    std::vector<const char*> cvals;
+    ckeys.reserve(elems.size());
+    cvals.reserve(elems.size());
+
+    for (int i=0; i < elems.size(); i++) {
+      std::vector<std::string> parts;
+      mxnet::kvstore::split(elems[i], ':', std::back_inserter(parts));
+      CHECK_NOTNULL(parts[0].c_str());
+      CHECK_NOTNULL(parts[1].c_str());
+      if (parts[0] == "filename") {
+        parts[1] = "rank" + std::to_string(ps::MyRank()) + "_" + parts[1];
+      }
+      char * ckey = new char[parts[0].length() + 1];
+      std::sprintf(ckey, "%s", parts[0].c_str());
+      ckeys.push_back(ckey);
+
+      char* cval = new char[parts[1].length() + 1];
+      std::sprintf(cval, "%s", parts[1].c_str());
```

Comment:
why sprintf? it’s not very efficient compared to a strcpy, which is generally pretty fast

Reply:
I want to call the C API, which expects arrays of char*  for keys and values. I had strcpy at first, but our lint complained that I should be using `sprintf` or `snprintf`. IMO efficiency is fine, because this is called once on at most 10 strings.

Reply:
lint asked me to switch from strcpy to snprintf or sprintf. IMO it's fine efficiency wise, because it will be called at most around 10 times on small strings, once for each of those strings.

I need char* arrays to call the C API